handler/handler.go

```diff
@@ -12,23 +13,34 @@ type CepHandler struct {
 	CepApis      map[string]api.Api
 }
 
-func (h *CepHandler) Handle(ctx *gin.Context) {
-	cep := ctx.Param("cep")
-	ctx.Header("Content-Type", "application/json; charset=utf-8")
+type responseError struct {
+	Error string `json:"error"`
+}
+
+func renderJSON(w http.ResponseWriter, code int, data interface{}) {
+	j, _ := json.Marshal(data)
+	w.WriteHeader(code)
+	w.Write(j)
+}
+
+func (h *CepHandler) Handle(w http.ResponseWriter, r *http.Request) {
+	cep := strings.Split(r.URL.Path, "/")[1]
```

Comment:
this has to be more defensive, it can trigger a panic that can make the server stop.. since we don't have an easy way to map path parameters from the URL, what about introducing a method like:

```go
cep, err := h.getCepFromUrl(r.URL)
if err != nil {
   renderJSON(w, http.StatusBadRequest, &resonseError{"Invalid cep provided"})
}

var invalidCep = errors.New("invalid cep provided")

func (h *CepHandler) getCepFromUrl(url url.URL) (string, error) {
   var path = r.URL.Path
   if len(path) <= 0 {
        return "", invalidCep
   }

   var pathParts strings.Split(path, "/")
   if len(pathParts) < 2 {
        return "", invalidCep
   }

   return pathParts[1]
}
```

and then please extend the tests to match the scenarios 👍